task_2/src/controllers/userController.ts

```diff
@@ -1,246 +1,111 @@
 import { Request, Response } from 'express'
-import fs from 'fs/promises'
 import { ValidationError } from 'joi'
-import { v4 as uuidv4 } from 'uuid'
-import { User, UserRequestBody, userRequestSchema } from '../models/User.js'
-import {
-  USER_CREATED_MESSAGE,
-  USER_UPDATED_MESSAGE,
-  USER_DELETED_MESSAGE,
-  SOMETHING_WENT_WRONG_MESSAGE,
-  NO_USER_FOUND_MESSAGE,
-  NO_LOGIN_SUBSTRING_MESSAGE,
-  NO_USERS_FOUND_MATCHING_LOGIN_SUBSTRING_MESSAGE,
-  LOGIN_NOT_AVAILABLE_MESSAGE,
-  databaseFilePath,
-} from './constants.js'
-import { jsonStringfy } from '../utils/jsonStringfy.js'
-import { sortUserByLogin } from '../utils/sortUserByLogin.js'
+import { UserRequestBody, userRequestSchema } from '../models/schema/userSchema.js'
+import { getGenericErrorMessage } from '../utils/index.js'
+import { UserService } from '../services/User.js'
+import { IUser } from '../models/User.js'
+
+const userService = new UserService()
 
 export const getUserById = async (req: Request, res: Response) => {
   try {
-    const userId = req.params.id
-    const user = await _getUserById(userId)
+    const userId = parseInt(req.params.id)
+    const result = await userService.getById(userId)
 
-    if (!user) {
-      return res.status(400).json({
-        success: false,
-        message: NO_USER_FOUND_MESSAGE,
-        error: userId,
-      })
+    if (result?.error) {
+      return res.status(400).json(result.error)
     }
 
-    res.status(200).json({
-      success: true,
-      data: user,
-    })
+    res.status(200).json(result.response)
```

Comment:
please check here for 404! result.response could be empty...

Reply:
Thanks for that, updated